src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -0,0 +1,305 @@
+import static com.coveros.selenified.utilities.Reporter.END_IDIV;
+
+/**
+ * Check will handle all verifications performed on the actual web services
+ * calls themselves. These asserts are custom to the framework, and in addition to
+ * providing easy object oriented capabilities, they assist in
+ * troubleshooting and debugging failing tests.
+ *
+ * @author Max Saperstone
+ * @version 3.2.0
+ * @lastupdate 4/4/2019
+ */
+public interface Check {
+
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    /**
+     * Retrieves the output file that we write all details out to
+     *
+     * @return Reporter
+     */
+    Reporter getReporter();
+
+    /**
+     * Retrieves the web services response  object, with all details
+     *
+     * @return Response
+     */
+    Response getResponse();
+
+    /**
+     * Casts an unknown JsonElement into an object, based on the expected behavior of it
+     *
+     * @param known   - what value are we expecting
+     * @param unknown - what JsonElement are we trying to cast
+     * @return Object - the expected object, properly cast
+     */
+    default Object castObject(Object known, JsonElement unknown) {
+        Object objectVal;
+        try {
+            if (known instanceof String) {
+                objectVal = unknown.getAsString();
+            } else if (known instanceof Integer) {
+                objectVal = unknown.getAsInt();
+            } else if (known instanceof Double) {
+                objectVal = unknown.getAsDouble();
+            } else if (known instanceof Float) {
+                objectVal = unknown.getAsFloat();
+            } else if (known instanceof Long) {
+                objectVal = unknown.getAsLong();
+            } else if (known instanceof Boolean) {
+                objectVal = unknown.getAsBoolean();
+            } else if (known instanceof Byte) {
+                objectVal = unknown.getAsByte();
+            } else if (known instanceof Character) {
+                objectVal = unknown.getAsCharacter();
+            } else if (known instanceof JsonArray) {
+                objectVal = unknown.getAsJsonArray();
+            } else if (known instanceof JsonObject) {
+                objectVal = unknown.getAsJsonObject();
+            } else {
+                objectVal = unknown;
+            }
+            return objectVal;
+        } catch (UnsupportedOperationException | NumberFormatException | IllegalStateException e) {
+            return unknown;
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // checks for our services
+    ///////////////////////////////////////////////////////////////////
+
+    /**
+     * Checks the actual response code is equals to the expected response
+     * code, and writes that out to the output file
+     *
+     * @param expectedCode - the expected response code
+     */
+    void equals(int expectedCode);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either override Object.equals(Object), or rename the method to prevent any confusion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1201)